server/src/main/java/org/opensearch/common/cache/LoadAwareCacheLoader.java

```diff
@@ -8,13 +8,16 @@
 
 package org.opensearch.common.cache;
 
+import org.opensearch.common.annotation.ExperimentalApi;
+
 /**
  * Extends a cache loader with awareness of whether the data is loaded or not.
  * @param <K> Type of key.
  * @param <V> Type of value.
  *
- * @opensearch.internal
+ * @opensearch.experimental
  */
+@ExperimentalApi
 public interface LoadAwareCacheLoader<K, V> extends CacheLoader<K, V> {
     boolean isLoaded();
 }
```

Comment:
Can this be moved into the `cache-common` module?

It's referenced in `OpenSearchOnHeapCache`, but that usage could just be `CacheLoader`.

Reply:
```OpenSearchOnHeapCache``` will need ```LoadAwareCacheLoader``` to calculate stats like below:
```
computeIfAbsent(K key, LoadAwareCacheLoader<K, V> loader) {
    V value = cache.computeIfAbsent(key, key1 -> loader.load(key));
    if (loader.isLoaded()) {
       // increment hit
    } else {
      // increment miss
    }
}
```

Reply:
Oh -- got it. Thanks!